pkg/plugin/datasource.go

```diff
@@ -173,8 +177,10 @@ func (d *Datasource) query(_ context.Context, _ backend.PluginContext, dataQuery
 		// new job, and we get isFinished=false.  When this is the case, grab the job ID and poll until the job is finished.
 		if !result.Header["isFinished"].(bool) {
 			elapsed := time.Since(startTime)
-			if elapsed >= MAX_DELAY_WAITING_FOR_FINISHED {
-				return backend.ErrDataResponse(backend.StatusBadRequest, fmt.Sprintf("Job %s still not finished after %v (status=%v)", jobId, elapsed, status))
+			// If there's a configured timeout, ensure we don't let the query run longer than that
+			if maxQueryDuration > 0 && elapsed >= maxQueryDuration {
+				// TODO: cancel the query
+				return backend.ErrDataResponse(backend.StatusBadRequest, fmt.Sprintf("Job %s still not finished after %v (status=%v)", jobId, maxQueryDuration, status))
```

Comment:
NOTE: The change to displaying `maxQueryDuration` instead of `elapsed` here just makes the error message cleaner.  If the timeout is 42s, but we determine this at 43.020007 seconds, it was displaying all weird.  This feels a bit more intuitive to show the configured timeout, a nice clean value.